tepeszCodes/document-management-system
# Chore: Setup unit testing for backend

#### What does this PR do?
Sets up unit tests, adds one sample test and moves logic responsible for listening to new requests to separate file (this was done to prevent having unresolved promises when testing)

#### Description of Task to be completed?
Setup unit testing for backend

#### How should this be manually tested?
Run `npm run test` script. All checks should pass. After that, change `res.send('This is my first Express server');` to `res.sendStatus(500);` in `app.js` and run `npm run test` one more time. Now the test should fail.

#### Any background context you want to provide?
@Avuidrauxs you didn't specified which test suites to use, so I decided to go with supertest and jest.

#### Screenshots (if appropriate)
![test passed](https://user-images.githubusercontent.com/54599709/129789858-46a93e7d-13bd-4d5e-806d-7c0ebd1f1576.png)

#### Questions:
Will we be working in TDD methodology?
